nrpy/infrastructures/BHaH/CurviBoundaryConditions/CurviBoundaryConditions.py

```diff
@@ -239,34 +248,34 @@ def Cfunction__EigenCoord_set_x0x1x2_inbounds__i0i1i2_inbounds_single_pt(
     body += (
         f"""
 // Step 1: Convert the (curvilinear) coordinate (x0,x1,x2) to Cartesian coordinates
-REAL xCart[3];  // where (x,y,z) is output
+{fp_type_alias} xCart[3];  // where (x,y,z) is output
```

Comment:
replace REAL with DOUBLE here and all other places.

Reply:
If you look at the new definition of `fp_type_alias`, it is `DOUBLE` only if `fp_type == "float"` otherwise it is `REAL`

Reply:
Sure, but here we want DOUBLE right? Also, removing f-string insertions makes the C code easier to modify in the future.

Reply:
That depends - I have not worked tested with higher precision.  I'm only enforcing `DOUBLE` for single precision with the expectation that `REAL = double, long, long double` would function as expected.

Reply:
It would be reasonable to just make DOUBLE a default macro in BHaH defines.  Currently I am adding this to supplemental defines in the nrpy example.

If you agree with this, I can include it in my future PR for BHaH_defines enhancements which is on hold until this PR is merged.

Reply:
I agree. Thanks @samueltootle

Reply:
I realized that the CIs would fail if `DOUBLE` wasn't added now, so I've included the modification to BHaH_defines just for the `DOUBLE` macro and made the change to specifically use `DOUBLE` instead of `{fp_type_alias}` as requested.

Reply:
Similar adjustments have now also been made for `conformally_flat_C_codegen_library.py`